spec/contracts/custom_fields/hierarchy/insert_item_contract_spec.rb

```diff
@@ -0,0 +1,119 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "rails_helper"
+
+RSpec.describe CustomFields::Hierarchy::InsertItemContract do
+  subject { described_class.new }
+
+  # rubocop:disable Rails/DeprecatedActiveModelErrorsMethods
+  describe "#call" do
+    let(:parent) { create(:hierarchy_item) }
+
+    context "when all required fields are valid" do
+      let(:params) { { parent:, label: "Valid Label" } }
+
+      it "is valid" do
+        result = subject.call(params)
+        expect(result).to be_success
+      end
+    end
+
+    context "when parent is not of type 'Item'" do
+      let(:invalid_parent) { create(:custom_field) }
+      let(:params) { { parent: invalid_parent, label: "Valid Label" } }
+
+      it "is invalid" do
+        result = subject.call(params)
+        expect(result).to be_failure
+        expect(result.errors.to_h).to include(parent: ["Parent must be of type 'Item'"])
+      end
+    end
+
+    context "when label is not unique within the same hierarchy level" do
+      before do
+        create(:hierarchy_item, parent:, label: "Duplicate Label")
+      end
+
+      let(:params) { { parent:, label: "Duplicate Label" } }
+
+      it "is invalid" do
+        result = subject.call(params)
+        expect(result).to be_failure
+        expect(result.errors.to_h).to include(label: ["Label must be unique within the same hierarchy level"])
+      end
+    end
+
+    context "when short is set and is a string" do
+      let(:params) { { parent:, label: "Valid Label", short: "Valid Short" } }
+
+      it "is valid" do
+        result = subject.call(params)
+        expect(result).to be_success
+      end
+    end
+
+    context "when short is set and is not a string" do
+      let(:params) { { parent:, label: "Valid Label", short: 123 } }
+
+      it "is invalid" do
+        result = subject.call(params)
+        expect(result).to be_failure
+        expect(result.errors.to_h).to include(short: ["must be a string"])
+      end
+    end
+
+    context "when inputs are valid" do
+      it "creates a success result" do
+        [
+          { parent:, label: "A label", short: "A shorthand" },
+          { parent:, label: "A label" }
+        ].each { |params| expect(subject.call(params)).to be_success }
+      end
+    end
+
+    context "when inputs are invalid" do
+      it "creates a failure result" do
+        [
+          { parent:, label: "A label", short: "" },
+          { parent:, label: "A label", short: nil },
```

Comment:
🟡 we might have a problem with that case. If I interpret the usage in the service correct, it will call it, when `short` is not passed.

Solutions:
a) adapt contract, so that passing `nil` is allowed, while `""` still is forbidden.
b) adapt service to call contract only with non-nil shorts or no short key at all.

To make b) work we would need to create a hash from the method parameters before putting them into the call. something like: 

```ruby
input = {parent:, label:, short:}.compact

CustomFields::Hierarchy::InsertItemContract
  .new
  .call(**input)
```

Reply:
`short` is indeed passed, having `short: nil` in the hash actually creates the key in the hash. Therefor you have three cases: `short: ""`, `short: nil` and no `short`.